### DISCUSSION TITLE
Is there a way to get the params for a particular rodauth route?

### DISCUSSION THREAD

**Discussion by donnovan-lyons** [janko/rodauth-rails#150] [Q&A]
I'm currently working with a Rails API backend, and I was looking for the right params to send via the `create-account` route , e.g. `login` and `password`, and I had a bit of difficulty. It wasn't until I downloaded the views that that I was able to see that it was via the id in the Rails form that the password confirmation params was `password-confirm` instead of `password_confirmation` or `password_confirm` or some other variant.

I am wondering if there is a way to better identify the params for a particular route, without having to rely on views, tests, etc.?

**Comment by janko** [comment]
Just as Rails engines such as Devise don't programmatically declare params their endpoints accept (e.g. in Rails routes), Rodauth too doesn't expose them. I find out about them by reading Rodauth's source code.

I think the best way to improve this would be to document default routes/params in official feature documentation. I've been playing with this idea for a while now, but haven't yet checked with Jeremy if and in which format would he accept this.
